### PR TITLE
Feature/sc 6933 switch to use wait helpers administration pages

### PR DESCRIPTION
**Pull request by alstanki**
# Description
Switched waitHelpers to use: Administration Pages

## Links to Tickets or other pull requests
https://ticketsystem.hpi-schul-cloud.org/browse/SC-6933

## Changes
Switch waitHelpers to use: Administration Pages
- prettierrc config
- waitAndClick -> click, clickAndWait

## Approval for review
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: page-objects/pages/administrationPages/ADMNSTRTNAdministerStudentsPage.js [hunk 1]
```diff
@@ -4,65 +4,97 @@ const startPage = require('../generalPagesBeforeLogin/StartPageBeforeLogin');
...
+	await waitHelpers.waitAndSetValue(lastNameInput, lastname);
+}
+
+async function setStudentEmail(email) {
+	await waitHelpers.waitAndSetValue(emailInput, email);
+}
+
+async function clickOnSendRegistrationLinkCheckbox() {
+	await elementHelpers.click(sendRegistrationLinkCheckbox);
+}
```
Comment (kgorner) [inline page-objects/pages/administrationPages/ADMNSTRTNAdministerStudentsPage.js:40]: in which cases should we use .click and in which clickAndWait?

File: page-objects/pages/administrationPages/ADMNSTRTNAdministerStudentsPage.js [hunk 2]
```diff
@@ -4,65 +4,97 @@ const startPage = require('../generalPagesBeforeLogin/StartPageBeforeLogin');
...
-            if (email === e_mail) {
-                let boxConsent = await driver.$(selectorNamesContainer + ' > tr:nth-child(' + i + ') > td:nth-child(7) > a:nth-child(2) > i');
-                await boxConsent.click();
-                let submitBtn = await driver.$(selectorConsentSubmitBtn);
-                let passwordField = await driver.$(passwordFieldSel);
-                let password_old = await passwordField.getValue();
-                oldPassword = password_old;
-                await submitBtn.click();
-                break;
+async function createNewPupil(firstname, lastname, email) {
```
Comment (kgorner) [inline page-objects/pages/administrationPages/ADMNSTRTNAdministerStudentsPage.js:45]: this function looks very nice and clean :-)

File: runtime/helpers/elementHelpers.js [hunk 3]
```diff
@@ -2,62 +2,75 @@
...
-				value: await opt.getAttribute("value"),
-			};
-		})
-	);
-};
+//Wait for element and click (without waitUntilPageLoads)
+async function click(selectorOrElement) {
+    const element = await waitHelpers.waitUntilElementIsClickable(selectorOrElement);
+    await element.click();
+    //waitUntilPageLoads(); is temporary:
```
Comment (kgorner) [inline runtime/helpers/elementHelpers.js:9]: in the end, if we wait for page load there is one difference between click() and clickAndWait - one more waitUntil page loads
=> do the tests run if we remove the waitUntilPageLoad in function click()?

**Review by kgorner — CHANGES_REQUESTED** [review]
at the end it is adding this empty line and maybe the removing of waitUntilPageLoad() in elementHelpers.click()
=> I like the code - we are making progress :-)

File: page-objects/pages/administrationPages/ADMNSTRTNAdministerClassesPage.js [hunk 4]
```diff
@@ -1,46 +1,67 @@
...
+//Administration: Classes
+async function verifyTitleOfAdministrationClassesPage() {
+    await waitHelpers.waitUntilPageTitleContains(titleOfAdministrationClassesPage)
+}
 
-        await waitHelpers.waitAndClick(classCreationExtraOptions);
-        await waitHelpers.waitAndSetValue(classNameInputField, className);
-        await waitHelpers.waitAndClick(confirmClassCreate);
+async function clickCreateClassBtn() {
+    await elementHelpers.clickAndWait(addClassBtn)
```
[the remark this answers is not in the document]
Comment (alstanki) [reply page-objects/pages/administrationPages/ADMNSTRTNAdministerClassesPage.js:22]: Now we have 2 different methods. clickAndWait() we should use when the page is reloaded or a new page is opened after clicking on the element. click() method we should use when we want to just click on an element without waitUntilPageLoads

File: page-objects/pages/administrationPages/ADMNSTRTNAdministerStudentsPage.js (hunk 1 above)
Comment (alstanki) [reply page-objects/pages/administrationPages/ADMNSTRTNAdministerStudentsPage.js:40]: Now we have 2 different methods. clickAndWait() we should use when the page is reloaded or a new page is opened after clicking on the element. click() method we should use when we want to just click on an element without waitUntilPageLoads

File: runtime/helpers/elementHelpers.js (hunk 3 above)
Comment (alstanki) [reply runtime/helpers/elementHelpers.js:9]: When I finish my current  task https://ticketsystem.hpi-schul-cloud.org/browse/SC-6771: Yes

File: page-objects/pages/administrationPages/ADMNSTRTNAdministerStudentsPage.js (hunk 1 above)
Comment (alstanki) [reply page-objects/pages/administrationPages/ADMNSTRTNAdministerStudentsPage.js:40]: Now we have 2 different methods. clickAndWait() we should use when the page is reloaded or a new page is opened after clicking on the element. click() method we should use when we want to just click on an element without waitUntilPageLoads. This PR is about changes in Administration Pages

File: page-objects/pages/administrationPages/ADMNSTRTNAdministerClassesPage.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (alstanki) [reply page-objects/pages/administrationPages/ADMNSTRTNAdministerClassesPage.js:22]: Now we have 2 different methods. clickAndWait() we should use when the page is reloaded or a new page is opened after clicking on the element. click() method we should use when we want to just click on an element without waitUntilPageLoads. This PR is about changes in Administration Pages. I have a lot of subtasks. See: SC-6771

File: page-objects/pages/administrationPages/ADMNSTRTNAdministerStudentsPage.js (hunk 2 above)
Comment (alstanki) [reply page-objects/pages/administrationPages/ADMNSTRTNAdministerStudentsPage.js:45]: tnx :)

File: .prettierrc.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+	singleQuote: true,
+	printWidth: 120,
+	useTabs: true,
+	trailingComma: "es5",
+};
```
[the remark this answers is not in the document]
Comment (alstanki) [reply .prettierrc.js:6]: Not needed-> It is by default for prettier prettier/prettier#6360

[the remark this answers is not in the document]
Comment (alstanki) [reply .prettierrc.js:6]: Not needed-> It is by default for prettier prettier/prettier#6360

File: features/course.feature [hunk 6]
```diff
@@ -64,8 +64,8 @@ Scenario Outline: logging in as a teacher I want to be able to edit a course on
     And teacher clicks on save changes button
     And teacher goes to courses page
     Then teacher should see that course with name <changeName> is displayed correctly on the list
-	Then teacher should see that course name <changeName> with description correctly displayed <description>
-	Then teacher should see that course name <changeName> with color correctly displayed <courseColour>
+	And teacher should see that course name <changeName> with description correctly displayed <description>
+	And teacher should see that course name <changeName> with color correctly displayed <courseColour>
```
[the remark this answers is not in the document]
Comment (alstanki) [reply features/course.feature:68]: :)

File: step_definitions/course-steps.js [hunk 7]
```diff
@@ -113,13 +122,13 @@ Then(/^.*clicks on save changes button$/, async function () {
...
 
-Then(/^.*should see that course name (.*) with description correctly displayed (.*)$/, async function (courseName, description) {
+Then(/^.*course name (.*) with description correctly displayed (.*)$/, async function (courseName, description) {
 	let index = await courseListPage.getIndexOfGivenCourseInSection(courseName, courseListPage.section.activeCourses)+1;
 	let courseDescriptionAfterChanges = await courseListPage.getDescriptionCourse(index);
 	expect(description).to.equal(courseDescriptionAfterChanges);
 });
 
-Then(/^.*should see that course name (.*) with color correctly displayed (.*)$/, async function (courseName, colorCourse) {
+Then(/^.*course name (.*) with color correctly displayed (.*)$/, async function (courseName, colorCourse) {
```
Comment (kgorner) [inline step_definitions/course-steps.js:131]: better: course name (.\*) with color (.\*) is displayed correctly

**Review by kgorner — APPROVED** [review]

File: step_definitions/course-steps.js (hunk 7 above)
Comment (alstanki) [reply step_definitions/course-steps.js:131]: My task concerns administration pages. I have only deleted what I thought was 100% redundant.

**Review by Grzesiek80 — APPROVED** [review]

**Review by kgorner — APPROVED** [review]

**Pull request merged**